samples/mpsl/timeslot/src/main.c

```diff
@@ -184,6 +195,14 @@ static void mpsl_nonpreemptible_thread(void)
 	/* Initialize to invalid session id */
 	mpsl_timeslot_session_id_t session_id = 0xFFu;
 
+#if defined(NRF5340_XXAA_NETWORK)
+	IRQ_DIRECT_CONNECT(SWI2_IRQn, 1, swi2_isr, 0);
+	irq_enable(SWI2_IRQn);
+#elif defined(NRF52840_XXAA) || defined(NRF52832_XXAA) || defined(NRF52832_XXAB)
+	IRQ_DIRECT_CONNECT(SWI2_EGU2_IRQn, 1, swi2_isr, 0);
+	irq_enable(SWI2_EGU2_IRQn);
+#endif
```

Comment:
Setting up the IRQs is related to the sample itself, it is not related to setting up the non-preemptible thread. Therefore I would suggest moving it to `main()`